design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,253 @@
+2. Once the resource has been created, changes to the clusters multisite configuration after the operator is restarted will not apply to the cluster. All further changes must be made manually through toolboxes.
+3. When the resource is deleted, realm, zonegroups, and master zones are not deleted from the cluster. Realm, zonegroup, and master zone deletion must be done through toolboxes.
+
+## Object Store Realm Walkthrough
+
+### Creating a Realm & Zonegroups
+
+#### Config
+
+When the storage admin is ready to create a realm and zonegroups for the cluster, the admin will specify his/her desired settings in the bootstrap-realm section on the configuration file.
+
+This example `object-store-realm.yaml`, names a realm with 2 zonegroups.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+  bootstrapRealm:
+    zonegroups:
+    - name: zonegroup-a
+      isMaster: true
+      masterZone: my-realm-zone-a
+    - name: zonegroup-b
+      isMaster: false
+      masterZone: my-realm-zone-b
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+
+At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will create all of the resources to start the object store realm.
+
+1. A Ceph realm is created if it does not exist already
+2. A Ceph master zonegroup is created in the realm if the realm does not exist already
+3. Any non-master Ceph zonegroups are created in the realm if the realm does not exist already
+4. A master zone is created for each zonegroup if the realm does not exist already.
+
+After these steps the admin should start up object stores configured with the same zonegroup, and realm as configured (see next section).
+
+#### Notes
+1. Once the resource has been created, changes to the config after the operator is restarted will not apply to the cluster. All further changes must be made manually through toolboxes.
+2. Once the resource has been created, changes to the clusters multisite configuration after the operator is restarted will not apply to the cluster. All further changes must be made manually through toolboxes.
+3. When the resource is deleted, realm, zonegroups, and master zones are not deleted from the cluster. Realm, zonegroup, and master zone deletion must be done through toolboxes.
+
+### Adding Object Stores to a Master Zone
+
+Once the storage admin creates a realm and zonegroup(s) on a cluster, the admin is ready to add object-stores to the master zones in a realm. 
+The admin will specify his/her configuration settings in the `realm` section of the object-store like in the following `object-store.yaml`.
+
+All objectstores wanting to be in a master zone must have the same name as a `masterZone` in the objects store realm config.
```

Comment:
This is not necessarily true as the objectstores zone could be changed to a master zone later on.